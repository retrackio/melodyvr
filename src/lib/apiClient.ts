import fetch from 'isomorphic-fetch';
import config from '../config';

export interface ReleasesResult {
  result: {
    releases: Release[];
  };
}

export interface Release {
  feature_order: number;
  is_free: boolean;
  product: Product;
}

export interface Product {
  id: number;
  title: string;
  description: string;
  image_hero_url: string;
  image_landscape_url: string;
  accent_colour_code: string;
  background_colour_code: string;
  text_colour_code: string;
  production_type_code: string;
  price: string | undefined;
}

export interface Venue {
  id: number;
  name: string;
  description: string;
  city: string;
  country_code: string;
}

export interface Artist {
  id: number;
  name: string;
  description: string;
  country_code: string;
  website_url: string;
  facebook_url: string;
  instagram_url: string;
  twitter_url: string;
  image_url: string;
  artist_background_colour_code: string;
  artist_text_colour_code: string;
  artist_accent_colour_code: string;
}

export interface Genre {
  id: number;
  name: string;
  description: string | null;
}

export type Performance = Product & {
  venue: Venue;
  artist: Artist;
  genre: Genre;
};

function sortReleases(releases: Release[]) {
  return releases.sort(
    (releaseA, releaseB) => releaseA.feature_order - releaseB.feature_order,
  );
}

export async function getReleases(): Promise<Release[]> {
  const response = await fetch(`${config.services.apiBaseUrl}/products`);
  const {
    result: { releases },
  } = await response.json();
  return sortReleases(releases);
}

export async function getPerformance(id: number): Promise<Performance> {
  const response = await fetch(`${config.services.apiBaseUrl}/products/${id}`);
  const { result } = await response.json();
  return result;
}
