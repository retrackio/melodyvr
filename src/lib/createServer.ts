import express from 'express';
import Server from 'next/dist/next-server/server/next-server';

export default function createServer(app: Server) {
  const server = express();
  const handle = app.getRequestHandler();

  server.all(
    '*',
    (req: express.Request, res: express.Response) => handle(req, res),
  );

  return server;
}
