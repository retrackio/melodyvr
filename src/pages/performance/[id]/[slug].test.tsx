import React from 'react';
import { NextPageContext } from 'next';
import { shallow } from 'enzyme';
import PerformancePage from './[slug]';
import { getPerformance } from '../../../lib/apiClient';

jest.mock('../../../lib/apiClient', () => ({
  getPerformance: jest.fn().mockResolvedValue({ fake: 'performance' }),
}));

const mockedPerformance: any = {
  title: 'Some performance',
};

describe('<Performance /> page', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <PerformancePage performance={mockedPerformance} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('getInitialProps', () => {
    it('returns the correct data', async () => {
      const mockContext = {
        query: {
          id: 99,
        },
      };
      const data = await PerformancePage.getInitialProps(
        (mockContext as any) as NextPageContext,
      );
      expect(data).toEqual({ performance: { fake: 'performance' } });
      expect(getPerformance).toHaveBeenCalledWith(99);
    });
  });
});
