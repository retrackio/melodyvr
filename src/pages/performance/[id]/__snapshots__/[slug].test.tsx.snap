// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Performance /> page renders correctly 1`] = `
<Fragment>
  <Head>
    <title>
      Performance - 
      Some performance
    </title>
    <meta
      content="Performance - Some performance"
      key="title"
      property="og:title"
    />
  </Head>
  <Wrapper>
    <PerformanceDetails
      performance={
        Object {
          "title": "Some performance",
        }
      }
    />
    <Link
      href="/"
    >
      <a>
        Back to homepage
      </a>
    </Link>
  </Wrapper>
</Fragment>
`;
